### DISCUSSION TITLE
Is there any ways to check whether a specific TextElement is supported by a font?

### DISCUSSION THREAD

**Discussion by OliveInto** [SixLabors/Fonts#437] [Q&A]
Currently I am using string.EnumerateRunes() to get runes in a string, and then using Font.TryGetGlyphs() to check GlyphType of the each Glyph generated by the corresponded rune.

However, in some situations, simply consider runes in a string is incomplete, I need to consider each TextElements by using StringInfo.GetTextElementEnumerator(). Since TryGetGlyphs cannot test TextElement, is there a better way?

**Comment by JimBobSquarePants** [comment]
You could enumerate each grapheme and then the code points related to each char in that grapheme?

```c#
namespace SixLabors.Fonts.Unicode;

SpanGraphemeEnumerator enumerator = new(text);
while (graphemeEnumerator.MoveNext())
{
    SpanCodePointEnumerator codePointEnumerator = new(graphemeEnumerator.Current);
    while (codePointEnumerator.MoveNext())
    {
        // Get the glyphs for each codepoint.
    }
}
```
